apps/veil/src/pages/tournament/api/use-epoch-results.ts

```diff
@@ -18,5 +31,47 @@ export const useEpochResults = (
 
   useRefetchOnNewBlock(name, query, disabled);
 
-  return query;
+  // collect all the gauges for the current epoch by the asset denom
+  const gaugeMapByDenom = useMemo(
+    () =>
+      (query.data?.data ?? []).reduce<Map<string, MappedGauge>>((accum, current) => {
+        accum.set(current.asset.base, current);
+        return accum;
+      }, new Map()),
+    [query.data?.data],
+  );
+
+  // filters assets from the registry that are IBC-assets and match the search query
+  const filteredAssets = useMemo<Metadata[]>(() => {
+    return assets.filter(asset => {
+      return (
+        assetPatterns.ibc.matches(asset.base) &&
+        (asset.symbol.toLowerCase().includes(search.toLowerCase()) ||
+          asset.description.toLowerCase().includes(search.toLowerCase()))
+      );
+    });
+  }, [assets, search]);
+
+  // adapts filtered assets to the gauges
+  const assetGauges = useMemo<MappedGauge[]>(() => {
+    return filteredAssets.map(asset => {
+      const fromGauge = gaugeMapByDenom.get(asset.base);
+      if (fromGauge) {
+        return fromGauge;
+      }
+
+      return {
+        asset,
+        epoch: 0,
+        votes: 0,
+        portion: 0,
+        missing_votes: 0,
+      };
+    });
```

Comment:
It's not correct to do this. The amount of missing votes is not 0. We should just not display assets that have received no votes.

Reply:
What do you mean? The field `missing_votes` is not used anywhere in the frontends, needed here only for typing reasons

Reply:
The issue with creating a meaningless value to satisfy the type checker, under the guise that the field isn't actually used, is that then someone subsequently ends up using the field, and now you have a bug.

In this case, probably fine to display 0 somewhere, but this was the source of an actual runtime error recently.

If we actually want to include a value here, the correct type is something like `MappedGauge | Asset`, actually representing the possibilities.

I think in this case, what we want is just to not consider assets which don't have gauge information available to them at all.

Reply:
I still don't see an issue here, but removed the `missing_votes` field from types and from the usage. 

> the correct type is something like MappedGauge | Asset

this would only bring confusion, and we'd still need mapping in some underlying component

> what we want is just to not consider assets which don't have gauge information available to them at all

we need it for UI/UX

Reply:
I think the actual situation here is that you have all the assets, some of which may have the information including the votes, their portion, etc. So the type you want is something like:
```ts
interface Example {
  asset: Asset;
  voteInfo?: MappedGauge | undefined;
}
```
but, eh, this isn't blocking for the PR.